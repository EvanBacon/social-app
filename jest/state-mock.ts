import {RootStoreModel} from './../src/state/models/root-store'
import {NavigationTabModel} from './../src/state/models/navigation'
import {SessionModel} from '../src/state/models/session'
import {NavigationModel} from '../src/state/models/navigation'
import {ShellUiModel} from '../src/state/models/shell-ui'
import {MeModel} from '../src/state/models/me'
import {OnboardModel} from '../src/state/models/onboard'
import {ProfilesViewModel} from '../src/state/models/profiles-view'
import {LinkMetasViewModel} from '../src/state/models/link-metas-view'
import {MembershipsViewModel} from '../src/state/models/memberships-view'
import {FeedModel} from '../src/state/models/feed-view'
import {NotificationsViewModel} from '../src/state/models/notifications-view'
import {LRUMap} from 'lru_map'

const mockedSessionStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  data: {
    service: '',
    refreshJwt: '',
    accessJwt: '',
    handle: '',
    did: '',
  },
  online: false,
  attemptingConnect: false,
  rootStore: {} as RootStoreModel,
  hasSession: true,
  clear: jest.fn(),
  setState: jest.fn(),
  setOnline: jest.fn(),
  updateAuthTokens: jest.fn(),
  connect: jest.fn(),
  describeService: jest.fn(),
  login: jest.fn(),
  createAccount: jest.fn(),
  logout: jest.fn(),

  // unknown added because of the missing private methods: _connectPromise, configureApi & _connect
} as unknown as SessionModel

const mockedNavigationTabStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  id: 0,
  history: [
    {
      url: '',
      ts: 0,
      title: '',
      id: 0,
    },
  ],
  index: 0,
  isNewTab: false,
  current: {
    url: '',
    ts: 0,
    title: '',
    id: 0,
  },
  canGoBack: false,
  canGoForward: false,
  backTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  forwardTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  navigate: jest.fn(),
  refresh: jest.fn(),
  goBack: jest.fn(),
  fixedTabReset: jest.fn(),
  goForward: jest.fn(),
  goToIndex: jest.fn(),
  setTitle: jest.fn(),
  setIsNewTab: jest.fn(),
  fixedTabPurpose: 0,
  getBackList: jest.fn(),
  getForwardList: jest.fn(),
} as NavigationTabModel

const mockedNavigationStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  tabs: [],
  tabIndex: 0,
  clear: jest.fn(),
  tab: mockedNavigationTabStore,
  tabCount: 1,
  isCurrentScreen: jest.fn(),
  navigate: jest.fn(),
  refresh: jest.fn(),
  setTitle: jest.fn(),
  handleLink: jest.fn(),
  switchTo: jest.fn(),
  setActiveTab: jest.fn(),
  closeTab: jest.fn(),
  newTab: jest.fn(),
} as NavigationModel

const mockedShellStore = {
  minimalShellMode: false,
  isMainMenuOpen: false,
  isModalActive: false,
  activeModal: undefined,
  isLightboxActive: false,
  activeLightbox: undefined,
  isComposerActive: false,
  composerOpts: undefined,
  setMainMenuOpen: jest.fn(),
  setMinimalShellMode: jest.fn(),
  openModal: jest.fn(),
  closeModal: jest.fn(),
  closeComposer: jest.fn(),
  closeLightbox: jest.fn(),
  openComposer: jest.fn(),
  openLightbox: jest.fn(),
} as ShellUiModel

const mockedMeStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  did: '',
  handle: '',
  displayName: '',
  description: '',
  avatar: '',
  notificationCount: 0,
  rootStore: {} as RootStoreModel,
  // TODO: mock these models
  memberships: {
    placeholder: null,
  } as MembershipsViewModel,
  mainFeed: {
    placeholder: null,
  } as FeedModel,
  notifications: {
    placeholder: null,
  } as NotificationsViewModel,
  clear: jest.fn(),
  load: jest.fn(),
  clearNotificationCount: jest.fn(),
  fetchStateUpdate: jest.fn(),
  refreshMemberships: jest.fn(),
} as MeModel

const mockedOnboardStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  isOnboarding: false,
  stage: '',
  start: jest.fn(),
  stop: jest.fn(),
  next: jest.fn(),
} as OnboardModel

const mockedProfileStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getProfile: jest.fn(),
  overwrite: jest.fn(),
} as ProfilesViewModel

const mockedLinkMetasStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getLinkMeta: jest.fn(),
} as LinkMetasViewModel

export const mockedRootStore = {
  api: false,
  resolveName: jest.fn(),
  serialize: jest.fn(),
  hydrate: jest.fn(),
  fetchStateUpdate: jest.fn(),
  clearAll: jest.fn(),
  session: mockedSessionStore,
  nav: mockedNavigationStore,
  shell: mockedShellStore,
  me: mockedMeStore,
  onboard: mockedOnboardStore,
  profiles: mockedProfileStore,
  linkMetas: mockedLinkMetasStore,
}
