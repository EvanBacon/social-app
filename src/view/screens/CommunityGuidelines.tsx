import React from 'react'
import {View} from 'react-native'
import {useFocusEffect} from '@react-navigation/native'
import {ViewHeader} from '../com/util/ViewHeader'
import {useStores} from 'state/index'
import {ScrollView} from 'view/com/util/Views'
import {Text} from 'view/com/util/text/Text'
import {usePalette} from 'lib/hooks/usePalette'
import {s} from 'lib/styles'
import Html from '../../locale/en/community-guidelines'

export const CommunityGuidelinesScreen = () => {
  const pal = usePalette('default')
  const store = useStores()

  useFocusEffect(
    React.useCallback(() => {
      store.shell.setMinimalShellMode(false)
    }, [store]),
  )

  return (
    <View>
      <ViewHeader title="Community Guidelines" />
      <ScrollView style={[s.hContentRegion, pal.view]}>
        <View style={[s.p20]}>
          <Text type="title-xl" style={[pal.text, s.bold, s.pb20]}>
            Community Guidelines
          </Text>
          <Html />
        </View>
        <View style={s.footerSpacer} />
      </ScrollView>
    </View>
  )
}
