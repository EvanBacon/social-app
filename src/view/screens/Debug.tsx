import React from 'react'
import {ScrollView, View} from 'react-native'
import {ViewHeader} from '../com/util/ViewHeader'
import {Text} from '../com/util/Text'
import {ThemeProvider, useTheme} from '../lib/ThemeContext'
import {PaletteColorName} from '../lib/ThemeContext'
import {usePalette} from '../lib/hooks/usePalette'

import {ViewSelector} from '../com/util/ViewSelector'
import {Button} from '../com/util/forms/Button'
import {ToggleButton} from '../com/util/forms/ToggleButton'
import {RadioGroup} from '../com/util/forms/RadioGroup'
import {ErrorScreen} from '../com/util/error/ErrorScreen'
import {ErrorMessage} from '../com/util/error/ErrorMessage'

const MAIN_VIEWS = ['Base', 'Controls', 'Error']

export const Debug = () => {
  const [colorScheme, setColorScheme] = React.useState<'light' | 'dark'>(
    'light',
  )
  const [currentView, setCurrentView] = React.useState<number>(0)
  const onToggleColorScheme = () => {
    setColorScheme(colorScheme === 'light' ? 'dark' : 'light')
  }

  const renderItem = item => {
    return (
      <View>
        <View style={{paddingTop: 10, paddingHorizontal: 10}}>
          <ToggleButton
            type="default-light"
            onPress={onToggleColorScheme}
            isSelected={colorScheme === 'dark'}
            label="Dark mode"
          />
        </View>
        {item.currentView === 2 ? (
          <ErrorView key="error" />
        ) : item.currentView === 1 ? (
          <ControlsView key="controls" />
        ) : (
          <BaseView key="base" />
        )}
      </View>
    )
  }

  const items = [{currentView}]

  return (
    <ThemeProvider theme={colorScheme}>
      <View style={{flex: 1}}>
        <ViewHeader title="Debug panel" />
        <ViewSelector
          swipeEnabled
          sections={MAIN_VIEWS}
          items={items}
          renderItem={renderItem}
          onSelectView={setCurrentView}
        />
      </View>
    </ThemeProvider>
  )
}

function Heading({label}: {label: string}) {
  const theme = useTheme()
  return (
    <View style={{paddingTop: 10, paddingBottom: 5}}>
      <Text style={theme.typography.h3}>{label}</Text>
    </View>
  )
}

function BaseView() {
  return (
    <View style={{paddingHorizontal: 10}}>
      <Heading label="Palettes" />
      <PaletteView palette="default" />
      <PaletteView palette="primary" />
      <PaletteView palette="secondary" />
      <PaletteView palette="error" />
      <Heading label="Typography" />
      <TypographyView />
      <View style={{height: 200}} />
    </View>
  )
}

function ControlsView() {
  return (
    <ScrollView style={{paddingHorizontal: 10}}>
      <Heading label="Buttons" />
      <ButtonsView />
      <Heading label="Toggle Buttons" />
      <ToggleButtonsView />
      <Heading label="Radio Buttons" />
      <RadioButtonsView />
      <View style={{height: 200}} />
    </ScrollView>
  )
}

function ErrorView() {
  return (
    <View style={{padding: 10}}>
      <View style={{marginBottom: 5}}>
        <ErrorScreen
          title="Error screen"
          message="A major error occurred that led the entire screen to fail"
          details="Here are some details"
          onPressTryAgain={() => {}}
        />
      </View>
      <View style={{marginBottom: 5}}>
        <ErrorMessage message="This is an error that occurred while things were being done" />
      </View>
      <View style={{marginBottom: 5}}>
        <ErrorMessage
          message="This is an error that occurred while things were being done"
          numberOfLines={1}
        />
      </View>
      <View style={{marginBottom: 5}}>
        <ErrorMessage
          message="This is an error that occurred while things were being done"
          onPressTryAgain={() => {}}
        />
      </View>
      <View style={{marginBottom: 5}}>
        <ErrorMessage
          message="This is an error that occurred while things were being done"
          onPressTryAgain={() => {}}
          numberOfLines={1}
        />
      </View>
    </View>
  )
}

function PaletteView({palette}: {palette: PaletteColorName}) {
  const theme = useTheme()
  const defaultPal = usePalette('default')
  const pal = usePalette(palette)
  return (
    <View
      style={[
        pal.view,
        pal.border,
        {
          padding: 10,
          marginBottom: 5,
        },
      ]}>
      <Text style={[theme.typography.body1, pal.text]}>{palette} colors</Text>
      <Text style={[theme.typography.body1, pal.textLight]}>Light text</Text>
      <Text style={[theme.typography.body1, pal.link]}>Link text</Text>
      {palette !== 'default' && (
        <View style={[defaultPal.view]}>
          <Text style={[theme.typography.body1, pal.textInverted]}>
            Inverted text
          </Text>
        </View>
      )}
    </View>
  )
}

function TypographyView() {
  const theme = useTheme()
  const pal = usePalette('default')
  return (
    <View
      style={[
        pal.view,
        pal.border,
        {
          marginBottom: 5,
          padding: 5,
        },
      ]}>
      <Text style={[pal.text, theme.typography.h1]}>Heading 1</Text>
      <Text style={[pal.text, theme.typography.h2]}>Heading 2</Text>
      <Text style={[pal.text, theme.typography.h3]}>Heading 3</Text>
      <Text style={[pal.text, theme.typography.h4]}>Heading 4</Text>
      <Text style={[pal.text, theme.typography.subtitle1]}>Subtitle 1</Text>
      <Text style={[pal.text, theme.typography.subtitle2]}>Subtitle 2</Text>
      <Text style={[pal.text, theme.typography.body1]}>Body 1</Text>
      <Text style={[pal.text, theme.typography.body2]}>Body 2</Text>
      <Text style={[pal.text, theme.typography.button]}>Button</Text>
      <Text style={[pal.text, theme.typography.caption]}>Caption</Text>
      <Text style={[pal.text, theme.typography.overline]}>Overline</Text>
    </View>
  )
}

function ButtonsView() {
  const defaultPal = usePalette('default')
  const buttonStyles = {marginRight: 5}
  return (
    <View
      style={[
        defaultPal.view,
        defaultPal.border,
        {
          marginBottom: 5,
          padding: 5,
        },
      ]}>
      <View
        style={{
          flexDirection: 'row',
          marginBottom: 5,
        }}>
        <Button type="primary" label="Primary solid" style={buttonStyles} />
        <Button type="secondary" label="Secondary solid" style={buttonStyles} />
      </View>
      <View style={{flexDirection: 'row'}}>
        <Button
          type="primary-outline"
          label="Primary outline"
          style={buttonStyles}
        />
        <Button
          type="secondary-outline"
          label="Secondary outline"
          style={buttonStyles}
        />
      </View>
      <View style={{flexDirection: 'row'}}>
        <Button
          type="primary-light"
          label="Primary light"
          style={buttonStyles}
        />
        <Button
          type="secondary-light"
          label="Secondary light"
          style={buttonStyles}
        />
      </View>
      <View style={{flexDirection: 'row'}}>
        <Button
          type="default-light"
          label="Default light"
          style={buttonStyles}
        />
      </View>
    </View>
  )
}

function ToggleButtonsView() {
  const defaultPal = usePalette('default')
  const buttonStyles = {marginBottom: 5}
  const [isSelected, setIsSelected] = React.useState(false)
  const onToggle = () => setIsSelected(!isSelected)
  return (
    <View
      style={[
        defaultPal.view,
        defaultPal.border,
        {
          marginBottom: 5,
          padding: 5,
        },
      ]}>
      <ToggleButton
        type="primary"
        label="Primary solid"
        style={buttonStyles}
        isSelected={isSelected}
        onPress={onToggle}
      />
      <ToggleButton
        type="secondary"
        label="Secondary solid"
        style={buttonStyles}
        isSelected={isSelected}
        onPress={onToggle}
      />
      <ToggleButton
        type="primary-outline"
        label="Primary outline"
        style={buttonStyles}
        isSelected={isSelected}
        onPress={onToggle}
      />
      <ToggleButton
        type="secondary-outline"
        label="Secondary outline"
        style={buttonStyles}
        isSelected={isSelected}
        onPress={onToggle}
      />
      <ToggleButton
        type="primary-light"
        label="Primary light"
        style={buttonStyles}
        isSelected={isSelected}
        onPress={onToggle}
      />
      <ToggleButton
        type="secondary-light"
        label="Secondary light"
        style={buttonStyles}
        isSelected={isSelected}
        onPress={onToggle}
      />
      <ToggleButton
        type="default-light"
        label="Default light"
        style={buttonStyles}
        isSelected={isSelected}
        onPress={onToggle}
      />
    </View>
  )
}

const RADIO_BUTTON_ITEMS = [
  {key: 'default-light', label: 'Default Light'},
  {key: 'primary', label: 'Primary'},
  {key: 'secondary', label: 'Secondary'},
  {key: 'primary-outline', label: 'Primary Outline'},
  {key: 'secondary-outline', label: 'Secondary Outline'},
  {key: 'primary-light', label: 'Primary Light'},
  {key: 'secondary-light', label: 'Secondary Light'},
]
function RadioButtonsView() {
  const defaultPal = usePalette('default')
  const [rgType, setRgType] = React.useState('default-light')
  return (
    <View
      style={[
        defaultPal.view,
        defaultPal.border,
        {
          marginBottom: 5,
          padding: 5,
        },
      ]}>
      <RadioGroup
        type={rgType}
        items={RADIO_BUTTON_ITEMS}
        initialSelection="default-light"
        onSelect={setRgType}
      />
    </View>
  )
}
