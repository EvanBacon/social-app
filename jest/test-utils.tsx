import React from 'react'
import {render} from '@testing-library/react-native'
import {GestureHandlerRootView} from 'react-native-gesture-handler'
import {RootSiblingParent} from 'react-native-root-siblings'
import {SafeAreaProvider} from 'react-native-safe-area-context'
import {RootStoreModel, RootStoreProvider} from '../src/state'
import {mockedRootStore} from './state-mock'

const WrappedComponent = ({children}: any) => {
  return (
    // eslint-disable-next-line react-native/no-inline-styles
    <GestureHandlerRootView style={{flex: 1}}>
      <RootSiblingParent>
        <RootStoreProvider value={mockedRootStore as RootStoreModel}>
          <SafeAreaProvider>{children}</SafeAreaProvider>
        </RootStoreProvider>
      </RootSiblingParent>
    </GestureHandlerRootView>
  )
}

const customRender = (ui: any, options?: any) =>
  render(ui, {wrapper: WrappedComponent, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}
