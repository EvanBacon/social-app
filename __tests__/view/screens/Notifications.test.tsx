import React from 'react'
import {Notifications} from '../../../src/view/screens/Notifications'
import renderer from 'react-test-renderer'
import {render} from '../../../jest/test-utils'

describe('Notifications', () => {
  const mockedProps = {
    navIdx: [0, 0] as [number, number],
    params: {},
    visible: true,
  }
  it('matches snapshot', () => {
    const tree = renderer.create(<Notifications {...mockedProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('tests', () => {
    render(<Notifications {...mockedProps} />)
  })
})
