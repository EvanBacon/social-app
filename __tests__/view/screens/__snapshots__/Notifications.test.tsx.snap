// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
    testID="notificationsView"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingBottom": 6,
            "paddingHorizontal": 12,
            "paddingTop": 6,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 30,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 30,
            "opacity": 1,
            "width": 40,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={30}
          bbWidth={30}
          focusable={false}
          height={30}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 30,
                "width": 30,
              },
            ]
          }
          vbHeight={100}
          vbWidth={100}
          width={30}
        >
          <RNSVGGroup>
            <RNSVGDefs>
              <RNSVGLinearGradient
                gradient={
                  Array [
                    0,
                    -6862849,
                    1,
                    -16742913,
                  ]
                }
                gradientTransform={null}
                gradientUnits={0}
                name="grad"
                x1="0"
                x2="1"
                y1="0"
                y2="1"
              />
            </RNSVGDefs>
            <RNSVGCircle
              cx="50"
              cy="50"
              fill={
                Array [
                  1,
                  "grad",
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              r="50"
            />
            <RNSVGText
              content={null}
              dx={Array []}
              dy={Array []}
              fill={4294967295}
              font={
                Object {
                  "fontSize": "50",
                  "fontWeight": "bold",
                  "textAnchor": "middle",
                }
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  "50",
                ]
              }
              y={
                Array [
                  "67",
                ]
              }
            >
              <RNSVGTSpan
                content="T"
                dx={Array []}
                dy={Array []}
                font={Object {}}
                rotate={Array []}
                x={Array []}
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "baseline",
            "flexDirection": "row",
            "marginRight": "auto",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 0.15,
              },
              Object {
                "color": "#000000",
              },
            ]
          }
          testID="headerTitle"
        >
          Notifications
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f8f3f3",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "center",
            "opacity": 1,
            "width": 36,
          }
        }
      >
        <
          icon="plus"
          size={18}
          style={
            Object {
              "color": "#000000",
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f8f3f3",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "center",
            "marginLeft": 4,
            "opacity": 1,
            "width": 36,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={18}
          bbWidth={18}
          color={4278190080}
          fill="none"
          focusable={false}
          height={18}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="currentColor"
          strokeWidth={3}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "color": "#000000",
              },
              Object {
                "flex": 0,
                "height": 18,
                "width": 18,
              },
            ]
          }
          tintColor={4278190080}
          vbHeight={24}
          vbWidth={24}
          width={18}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              Array [
                2,
              ]
            }
            strokeWidth={3}
          >
            <RNSVGPath
              d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
              propList={
                Array [
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              strokeLinecap={1}
              strokeLinejoin={1}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f8f3f3",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "center",
            "marginLeft": 4,
            "opacity": 1,
            "width": 36,
          }
        }
      >
        <
          icon="signal"
          size={16}
          style={
            Object {
              "color": "#000000",
            }
          }
        />
        <
          icon="x"
          size={8}
          style={
            Object {
              "backgroundColor": "#f8f3f3",
              "bottom": 7,
              "color": "#ec4899",
              "position": "absolute",
              "right": 7,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        data={
          Array [
            Object {
              "_reactKey": "item-1",
              "author": Object {
                "declaration": Object {
                  "actorType": "",
                  "cid": "",
                },
                "did": "test did",
                "handle": "test.handle",
              },
              "cid": "",
              "copy": [MockFunction],
              "fetchAdditionalData": [MockFunction],
              "indexedAt": "",
              "isAssertion": false,
              "isFollow": false,
              "isInvite": false,
              "isMention": false,
              "isRead": true,
              "isReply": false,
              "isRepost": false,
              "isTrend": false,
              "isUpvote": true,
              "needsAdditionalData": false,
              "reason": "test reason",
              "rootStore": Object {},
              "subjectUri": "testuri",
              "toSupportedRecord": [MockFunction],
              "uri": "testuri",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onEndReached={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onRefresh={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        refreshControl={
          <RefreshControlMock
            onRefresh={[Function]}
            refreshing={false}
          />
        }
        refreshing={false}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        testID="feedFlatList"
        viewabilityConfigCallbackPairs={Array []}
      >
        <RCTRefreshControl />
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "borderTopWidth": 1,
                    "padding": 10,
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                  },
                  Object {
                    "borderColor": "#e4e2e2",
                  },
                  undefined,
                ]
              }
              testID="linkButton-/profile/testuri/post/"
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "width": 35,
                    }
                  }
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight={26}
                    bbWidth={26}
                    color={4293675161}
                    focusable={false}
                    height={26}
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Array [
                          Object {
                            "marginRight": 10,
                            "marginTop": 4,
                          },
                          Object {
                            "color": "#ec4899",
                          },
                          Object {
                            "position": "relative",
                            "top": -4,
                          },
                        ],
                        Object {
                          "flex": 0,
                          "height": 26,
                          "width": 26,
                        },
                      ]
                    }
                    tintColor={4293675161}
                    vbHeight={14}
                    vbWidth={14}
                    width={26}
                  >
                    <RNSVGGroup>
                      <RNSVGPath
                        d="M 7 3 L 2 8 L 4.5 8 L 4.5 11.5 L 9.5 11.5 L 9.5 8 L 12 8 L 7 3 Z"
                        fill={
                          Array [
                            2,
                          ]
                        }
                        propList={
                          Array [
                            "fill",
                            "stroke",
                            "strokeWidth",
                            "strokeLinecap",
                            "strokeLinejoin",
                          ]
                        }
                        stroke={
                          Array [
                            2,
                          ]
                        }
                        strokeLinecap={1}
                        strokeLinejoin={1}
                        strokeWidth={1.3}
                      />
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "marginRight": 3,
                          "opacity": 1,
                        }
                      }
                      testID="linkButton-/profile/test.handle"
                    >
                      <RNSVGSvgView
                        align="xMidYMid"
                        bbHeight={30}
                        bbWidth={30}
                        focusable={false}
                        height={30}
                        meetOrSlice={0}
                        minX={0}
                        minY={0}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            Object {
                              "flex": 0,
                              "height": 30,
                              "width": 30,
                            },
                          ]
                        }
                        vbHeight={100}
                        vbWidth={100}
                        width={30}
                      >
                        <RNSVGGroup>
                          <RNSVGDefs>
                            <RNSVGLinearGradient
                              gradient={
                                Array [
                                  0,
                                  -2424577,
                                  1,
                                  -6862849,
                                ]
                              }
                              gradientTransform={null}
                              gradientUnits={0}
                              name="grad"
                              x1="0"
                              x2="1"
                              y1="0"
                              y2="1"
                            />
                          </RNSVGDefs>
                          <RNSVGCircle
                            cx="50"
                            cy="50"
                            fill={
                              Array [
                                1,
                                "grad",
                              ]
                            }
                            propList={
                              Array [
                                "fill",
                              ]
                            }
                            r="50"
                          />
                          <RNSVGText
                            content={null}
                            dx={Array []}
                            dy={Array []}
                            fill={4294967295}
                            font={
                              Object {
                                "fontSize": "50",
                                "fontWeight": "bold",
                                "textAnchor": "middle",
                              }
                            }
                            propList={
                              Array [
                                "fill",
                              ]
                            }
                            rotate={Array []}
                            x={
                              Array [
                                "50",
                              ]
                            }
                            y={
                              Array [
                                "67",
                              ]
                            }
                          >
                            <RNSVGTSpan
                              content="T"
                              dx={Array []}
                              dy={Array []}
                              font={Object {}}
                              rotate={Array []}
                              x={Array []}
                              y={Array []}
                            />
                          </RNSVGText>
                        </RNSVGGroup>
                      </RNSVGSvgView>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "flexWrap": "wrap",
                        "paddingBottom": 2,
                        "paddingTop": 6,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                          "paddingRight": 3,
                        }
                      }
                      testID="linkButton-/profile/test.handle"
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                            },
                            Object {
                              "fontSize": 16,
                              "letterSpacing": 0.25,
                            },
                            Array [
                              Object {
                                "color": "#000000",
                              },
                              Object {
                                "fontWeight": "bold",
                              },
                            ],
                          ]
                        }
                      >
                        test.handle
                      </Text>
                    </View>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                          },
                          Object {
                            "fontSize": 16,
                            "letterSpacing": 0.25,
                          },
                          Array [
                            Object {
                              "paddingRight": 3,
                            },
                            Object {
                              "color": "#000000",
                            },
                          ],
                        ]
                      }
                    >
                      upvoted your post
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                          },
                          Object {
                            "fontSize": 16,
                            "letterSpacing": 0.25,
                          },
                          Array [
                            Object {
                              "paddingRight": 3,
                            },
                            Object {
                              "color": "#645454",
                            },
                          ],
                        ]
                      }
                    >
                      Invalid Date
                    </Text>
                  </View>
                  <View
                    testID="emptyAdditionalPost"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
